
For us the t2 wrapper is the compiler executable which is often rebuild (e.g.
on config change. So we most often want to prevent hashing the compiler (or
better our wrapper) size and date ...

(Originally written in ROCK Linux, commented and rediffed for recent
 ccache versions.)

  - Rene Rebe <rene@exactcode.de>

--- ccache-2.4/ccache.c	2004-09-13 12:38:30.000000000 +0200
+++ ccache-2.4-patched/ccache.c	2004-12-22 08:49:51.959011840 +0100
@@ -319,20 +319,22 @@
 
 	/* the compiler driver size and date. This is a simple minded way
 	   to try and detect compiler upgrades. It is not 100% reliable */
-	if (stat(args->argv[0], &st) != 0) {
+	if (!getenv("CCACHE_DONT_STAT_COMPILER")) {
+	  if (stat(args->argv[0], &st) != 0) {
 		cc_log("Couldn't stat the compiler!? (argv[0]='%s')\n", args->argv[0]);
 		stats_update(STATS_COMPILER);
 		failed();
-	}
+	  }
 
-	/* also include the hash of the compiler name - as some compilers
-	   use hard links and behave differently depending on the real name */
-	if (st.st_nlink > 1) {
+	  /* also include the hash of the compiler name - as some compilers
+	     use hard links and behave differently depending on the real name */
+	  if (st.st_nlink > 1) {
 		hash_string(str_basename(args->argv[0]));
-	}
+	  }
 
-	hash_int(st.st_size);
-	hash_int(st.st_mtime);
+	  hash_int(st.st_size);
+	  hash_int(st.st_mtime);
+	}
 
 	/* possibly hash the current working directory */
 	if (getenv("CCACHE_HASHDIR")) {
